src/vulkan-renderer/world/indentation.cpp

```diff
@@ -0,0 +1,74 @@
+#include "inexor/vulkan-renderer/world/indentation.hpp"
+
+#include <algorithm>
+#include <array>
+#include <cassert>
+#include <cmath>
+
+namespace inexor::vulkan_renderer::world {
+Indentation::Indentation(const std::uint8_t start, const std::uint8_t end) noexcept : m_start(start), m_end(end) {}
+
+Indentation::Indentation(const std::uint8_t uid) noexcept {
+    assert(uid <= 44);
+    constexpr std::array<std::uint8_t, Indentation::MAX> masks{44, 42, 39, 35, 30, 24, 17, 9};
+    for (std::uint8_t idx = 0; idx < Indentation::MAX; idx++) {
+        if (masks[idx] <= uid) {
+            m_start = Indentation::MAX - idx;
+            m_end = m_start + (uid - masks[idx]);
+            return;
+        }
+    }
+    m_start = 0;
+    m_end = uid;
+}
+
+bool Indentation::operator==(const Indentation &rhs) const {
+    return this->m_start == rhs.m_start && this->m_end == rhs.m_end;
+}
+
+bool Indentation::operator!=(const Indentation &rhs) const {
+    return !(*this == rhs);
+}
```

Comment:
Maybe a todo to use `operator<=>`could be added here for when (if) we move to c++ 20?